### DISCUSSION TITLE
User Defined Geocaching Platform (UDGP)

### DISCUSSION THREAD

**Discussion by baiti** [cgeo/cgeo#9546] [General]
[This discussion](https://github.com/cgeo/cgeo/discussions/9544) leads me to a more general perspective. cgeo  today interacts with and supports multiple Geocaching Platforms (GP). Most notably geocaching.com but also opencaching.*.  The GPs can be configured and activated via Settings -> Services - > Geocaching Platform. There is an entry in the list for each supported platform.

I would like to suggest to provide the capability to interoperate with a "User Defined Geocaching Platform". We would need to sit down and come up with an architecture that defines a relatively thin REST API that the UDGP has to provide in order to seamlessly integrate with cgeo. Since cgeo exists and UDGP does not, we would perhaps define the API such that it fits into cgeo with the least amount of friction possible.

I would anticipate that at some point the time is ready for a new GP. For various reasons which are most likely obvious to the casual observer. Defining and implementing a small, compact UDGP is not an insurmountable task by itself. On the other hand what is an insurmountable task for all practical purposes is to write a front end like cgeo.

The only realistic possibility for a UDGP is to be able to integrate with cgeo. Everything else is completely unrealistic.

**Comment by baiti** [comment]
May I interpret the silence on this thread in the sense that I have expressed the tought in a too complicated way?

I can simplify it significantly: It is about providing an additional "connector". The connectors that exist today are:

geocaching.com
openaching.{de,uk, pl, ...}
extremcacing.com

I am suggesting to add an additinoal one and just give it the name "User Defined Geocacching Platform".  When clicked, it should open a dialog where I can enter an URL (the endpoint of my REST API) and an access_token. The connector then consumes my REST API for the usual functions.

**Comment by Lineflyer** [comment]
You could also fork c:geo and use it (under same license) for connecting your own platform.
I would prefer only to integrate a service once its ready and publicly available.

Please also be reminded that all parts of c:geo needs to be maintained and tested. Therefore it should be well considered for what amount of users such function might be helpful or wanted.

**Reply by baiti** [reply]
I understand both aspects you mentioned. Fork and use would certainly be a possibility, but clearly not one I'd ever pursue.

Having a service ready and publicly available as an up front requirement kills any innovation. Traditionaly things start very small with perhaps just a handful of users and a bare minumum set of functions and then as time passes there will be more functions and more users.

How did extremcaching.com get into the picture? it seems to be around since 2015, yet on the webpage a "now in Beta Phase" is mentioned. After at least 5 years It is not publicly available, it works "by invite only". I was wondering, how many active users does it have?

How did it convince the thought leaders at cgeo to get their buy in? I am asking these questions as I might try to pursue the same routes then in the sense that what worked once might also work again.

**Comment by Lineflyer** [comment]
Extremcaching.com started with open user registration and the developers contributed the initial implementation into c:geo.
However since then the status seems not the best. We even thought about removing the interface as it had SSL problems over a rather long time. However at this moment it works normal for quite some time.

**Reply by baiti** [reply]
I see that this is not really an encouraging example. Let me think about it a bit more in order to come to some senses whether it would be really worthwhile to start something like that. With respect to an initial implementation, I had assumed that I would do that too, of course.

It would be a very interesting discussion to share what leads me to these thoughts but most likely not appropriate for this channel.

**Reply by Lineflyer** [reply]
> It would be a very interesting discussion to share what leads me to these thoughts but most likely not appropriate for this channel.

Well, these discussions can of course be also offtopic or more wider. Go for it if you want.
Not sure though how many interaction we will have here, but I opened it up to allow such kind of discussion (for those interested) rather than have it in our issues (as these should stay closely on-topic IMHO).

**Reply by baiti** [reply]
> > It would be a very interesting discussion to share what leads me to these thoughts but most likely not appropriate for this channel.
> 
> Well, these discussions can of course be also offtopic or more wider. Go for it if you want.
> Not sure though how many interaction we will have here, but I opened it up to allow such kind of discussion (for those interested) rather than have it in our issues (as these should stay closely on-topic IMHO).

Yes, that resonates well with me and it is exactly why I didn't want to go the "issues route".
